Shared/Supporting Files/Views/CategoryView.swift

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Esri
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import SwiftUI
+
+struct CategoryView: View {
+    /// A Boolean value that indicates whether the user is searching.
+    @Environment(\.isSearching) private var isSearching
+    
+    /// All samples retrieved from the Samples directory.
+    let samples: [Sample]
+    
+    /// The search query in the search bar.
+    @Binding var query: String
+    
+    /// A Boolean value that indicates whether to present the about view.
+    @State private var isAboutViewPresented = false
+    
+    /// The samples to display in the search list. Searching adjusts this value.
+    private var displayedSamples: [Sample] {
+        searchSamples()
```

Comment:
As a placeholder it is fine. I was thinking sth like this

```swift
    private var displayedSamples: [Sample] {
        if query.isEmpty {
            return []
        } else {
            return samples.filter { sample in
                // Can use better method names
                conditionName(sample.name, query) ||
                conditionCategory(sample.category, query) ||
                conditionTags(sample.tags, query)
            }
        }
    }
```

---

The other thing I was thinking is that probably the search result (displayedSamples) can also include the "source" of the result. Sth like 

```
Search term:

map

Result:

Display map
Display a map with imagery basemap.
matches: title, description, tags
```

This can be a follow up task in the future.

Reply:
I already have a prototype for the search working, but I was saving it for a future PR. Is this placeholder good enough for this PR, or do you want me to update it?

Reply:
Yes, let's do it in another PR.